File: OCH_Tao/ABC/C/339.js
const input = (require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n"));
const N = parseInt(input[0], 10);
const A = (input[1].split(" ")).map(x => parseInt(x, 10));
let cnt = 0;
for (let i = 0; i < N; i++) {
  cnt += A[i];
  if (cnt < 0) {
    cnt = 0;
  }
}
console.log(cnt);

File: OCH_Tao/ABC/B/339.py
H,W,N = map(int,input().split())
l = [["."]*W for i in range(H)]
x = 0
y = 0
f = 0
for i in range(N):
  if l[x][y]==".":
    l[x][y]="#"
    f += 1
  else:
    l[x][y]="."
    f -= 1
  f %= 4
  if f == 0:
    x -= 1
    x %= H
  elif f == 1:
    y += 1
    y %= W
  elif f == 2:
    x += 1
    x %= H
  else:
    y -= 1
    y %= W
for i in range(H):
  print("".join(l[i]))